web/interceptor/tokens_test.go

```diff
@@ -0,0 +1,159 @@
+package interceptor_test
+
+import (
+	"context"
+	"net"
+	"testing"
+	"time"
+
+	"github.com/golang-jwt/jwt"
+	"github.com/quickfeed/quickfeed/ci"
+	"github.com/quickfeed/quickfeed/internal/qtest"
+	"github.com/quickfeed/quickfeed/qf"
+	"github.com/quickfeed/quickfeed/qlog"
+	"github.com/quickfeed/quickfeed/scm"
+	"github.com/quickfeed/quickfeed/web"
+	"github.com/quickfeed/quickfeed/web/auth"
+	"github.com/quickfeed/quickfeed/web/interceptor"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+	"google.golang.org/grpc/test/bufconn"
+)
+
+func TestRefreshTokens(t *testing.T) {
+	db, cleanup := qtest.TestDB(t)
+	defer cleanup()
+	logger := qlog.Logger(t)
+	ags := web.NewQuickFeedService(logger.Desugar(), db, scm.TestSCMManager(), web.BaseHookOptions{}, &ci.Local{})
+
+	tm, err := auth.NewTokenManager(db, "test")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	lis := bufconn.Listen(BufSize)
+	bufDialer := func(context.Context, string) (net.Conn, error) {
+		return lis.Dial()
+	}
+	opt := grpc.ChainUnaryInterceptor(
+		interceptor.UnaryUserVerifier(logger, tm),
+		interceptor.TokenRefresher(logger, tm),
+	)
+	s := grpc.NewServer(opt) // skipcq: GO-S0902
+	qf.RegisterQuickFeedServiceServer(s, ags)
+
+	go func() {
+		if err := s.Serve(lis); err != nil {
+			t.Errorf("Server exited with error: %v", err)
+			return
+		}
+	}()
+
+	admin := qtest.CreateFakeUser(t, db, 1)
```

Comment:
These two should be moved below `client` on line 62 since they are not used until that time... Ref also to #732 (to be done in a separate PR later).